src/mem/ruby/protocol/SConscript

```diff
@@ -91,6 +91,14 @@ def slicc_emitter(target, source, env):
 
         files.update([output_dir.File(f) for f in sorted(slicc.files())])
 
+        # Dynamically determine protocol and add ProtocolInfo.hh to the list of
+        # files to be built
+        protocol_name = os.path.splitext(os.path.basename(filepath))[0]
+        protocol_file = output_dir.File(
+            f"{protocol_name}/{protocol_name}ProtocolInfo.hh"
+        )
+        files.update([protocol_file])
```

Comment:
are you sure there isn't a parameter or something you can access in scons that can tell you what these particles are instead of doing this string parsing of the paths?

Reply:
I updated the code to use `slicc.protocol` variable.